File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Java/Main.java
public class Main {
    public static void main(String[] args) {
        // Create handlers
        SupportHandler basic = new BasicSupportHandler();
        SupportHandler supervisor = new SupervisorSupportHandler();
        SupportHandler manager = new ManagerSupportHandler();

        // Chain them together
        basic.setNextHandler(supervisor);
        supervisor.setNextHandler(manager);

        // Send different types of requests
        SupportRequest r1 = new SupportRequest("basic");
        SupportRequest r2 = new SupportRequest("intermediate");
        SupportRequest r3 = new SupportRequest("advanced");
        SupportRequest r4 = new SupportRequest("unknown");

        basic.handleRequest(r1);  // Handled by Basic
        basic.handleRequest(r2);  // Handled by Supervisor
        basic.handleRequest(r3);  // Handled by Manager
        basic.handleRequest(r4);  // Unhandled
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Cpp/main.cpp
#include "./include/SupportRequest.h"
#include "./include/BasicSupportHandler.h"
#include "./include/SupervisorSupportHandler.h"
#include "./include/ManagerSupportHandler.h"

int main() {
    // Create handlers
    BasicSupportHandler basic;
    SupervisorSupportHandler supervisor;
    ManagerSupportHandler manager;

    // Set chain
    basic.setNextHandler(&supervisor);
    supervisor.setNextHandler(&manager);

    // Create and process requests
    SupportRequest r1("basic");
    SupportRequest r2("intermediate");
    SupportRequest r3("advanced");
    SupportRequest r4("unknown");

    basic.handleRequest(r1);
    basic.handleRequest(r2);
    basic.handleRequest(r3);
    basic.handleRequest(r4);

    return 0;
}
